src/runtime/ArrayBufferObject.h

```diff
@@ -53,22 +54,24 @@ class ArrayBufferObject : public Object {
     static const uint32_t maxArrayBufferSize = 210000000;
 
     void allocateBuffer(ExecutionState& state, size_t bytelength);
-    void attachBuffer(ExecutionState& state, void* buffer, size_t bytelength);
-    void attachExternalBuffer(ExecutionState& state, void* buffer, size_t bytelength);
+    void attachBuffer(BackingStore* backingStore);
     void detachArrayBuffer();
+    Optional<BackingStore*> backingStore()
+    {
+        if (m_backingStore) {
+            m_backingStore->m_isShared = true;
```

Comment:
I have a question about `m_isShared` variable set at here.
If `m_isShared` value represents that this BackingStore is allocated for ArrayBuffer object, then `m_isShared` already has a true value. So, I think that this line is unnecessary. What do you think about this?

Reply:
Oh. you are right.
I would not remove m_mayPointsSharedBackingStore variable.